### DISCUSSION TITLE
Query data error when using self signed certificate

### DISCUSSION THREAD

**Discussion by arminus** [grafana/grafana-infinity-datasource#305] [Q&A] [answered]
I'm trying to load a CSV from a spring boot endpoint but all I'm getting is a 

{"status":500,"statusText":"Internal Server Error","data":{"message":"Query data error"} ... (full error below)

If I paste the URL into postman on the same box, the CSV file loads fine. Manually downloading the file and serving it with an ad-hoc server like pupcloud and loading it from there into grafana-infinity works. In postman, the responses look identical except for a different Content-Type header. The spring boot endpoint sets application/octet-stream, while the test server sets application/vnd.ms-excel

Could the 500 be caused by that application/octet-stream content-type header? The only other difference is that in dev mode, the springboot endpoint is served with a self-signed cert on https - but that shouldn't cause a 500 in case it's not accepted? I also tried setting the params separately, no difference.

`
{"status":500,"statusText":"Internal Server Error","data":{"message":"Query data error"},"config":{"url":"api/ds/query","method":"POST","data":{"queries":[{"refId":"A","key":"Q-aa7bc20f-572a-4d25-8f8e-041f2a0e694a-0","type":"csv","source":"url","format":"table","url":"https://dev.local/file?filePath=/data/my.csv","url_options":{"method":"GET","data":"","params":[]},"root_selector":"","columns":[],"filters":[],"global_query_id":"","datasource":{"uid":"_Ap3WPw7k","type":"yesoreyeram-infinity-datasource"},"datasourceId":2,"intervalMs":60000,"maxDataPoints":1128}],"range":{"from":"2017-09-30T22:00:00.000Z","to":"2017-10-01T21:59:59.000Z","raw":{"from":"2017-09-30T22:00:00.000Z","to":"2017-10-01T21:59:59.000Z"}},"from":"1506808800000","to":"1506895199000"},"requestId":"explore_left","retry":0,"headers":{"X-Grafana-Org-Id":1},"hideFromInspector":false},"message":"Query data error"}
`

**Comment by yesoreyeram** [comment, marked as answer]
Have you tried the Skip TLS verify settings? ( for self signed certificates, you may need to skip TLS verification or provide appropriate CA cert to use.

<img width="738" alt="image" src="https://user-images.githubusercontent.com/153843/165359866-32d23053-f7ec-44e1-a053-52ff97b845e8.png">

**Reply by arminus** [reply]
Thanks, that was too easy. The 500 error through me off I guess...
